pkg/sinks/sink.go

```diff
@@ -187,7 +219,23 @@ func (u *SinkProcessor) Start(ctx context.Context) error {
 			}
 		}
 
-		// NOTE: forwarder can make multiple sink writers when we introduce fallback sinks
+		// create the main sink writer
+		sinkWriter, err := u.createSinkWriter(ctx, &u.VertexInstance.Vertex.Spec.Sink.AbstractSink, sinkHandler)
+		if err != nil {
+			return fmt.Errorf("failed to find a sink, error: %w", err)
+		}
+
+		// create the fallback sink writer if fallback sink is present
+		fbSink := u.VertexInstance.Vertex.Spec.Sink.Fallback
+		if fbSink != nil && fbSink.IsAnySinkSpecified() {
+			fbSinkWriter, err := u.createSinkWriter(ctx, u.VertexInstance.Vertex.Spec.Sink.Fallback, fbSinkHandler)
```

Comment:
Who handles the close of `gbSinkWriter`?

Reply:
Instead of using an option to pass in the fallback writer, is it better to explicitly add it as a property to the forwarder?

Reply:
Good find... The forwarder should close it, I missed that.

Reply:
fb sink writer is not a mandatory field for the forwarder to start, so kept it optional. Since it is optional for users as well